src/actions/dashboards.js

```diff
@@ -71,56 +72,55 @@ export const getPGELoadProfile = (startDate, endDate) => (dispatch) => {
       const date = data[''];
       if (date) {
         delete data[''];
-        loadProfile.set(new Date(date), data);
+        loadProfile.set(date, data);
       }
     })
     .on('end', () => {
-      dispatch(getPGELoadProfileCompleted(loadProfile));
+      dispatch(getPGELoadProfileCompleted(loadProfile, addChartModalVisible));
     });
 
   const start = moment(startDate);
   const stop = moment(endDate);
-  const stopFormatted = stop.format('YYYYMMDD');
 
-  for (let date = start; date.isSameOrBefore(stop); date.add(1, 'days')) {
-    /*    const dateStringKey = `PGE${date.format('YYYYMMDD')}`;
+  const loadProfileCsvs = new SortedMap([], (a, b) => a - b);
+
+  let waitingOnAPI = false;
+
+  for (let date = stop; date.isSameOrAfter(start); date.subtract(1, 'days')) {
+    const dateStringKey = `PGE${date.format('YYYYMMDD')}`;
 
     const localLoadProfile = window.localStorage.getItem(dateStringKey);
 
     if (localLoadProfile) {
-      try {
-        PGELoadProfileStream.write(localLoadProfile);
-        if (date.isSame(stop, 'day')) {
-          dispatch(getPGELoadProfileCompleted(loadProfile));
-        }
-      } catch (err) {
-        dispatch(getPGELoadProfileFailed(err));
-      }
-    } else { */
-    ApiClient.get(`/api/pge/${date.format('YYYYMMDD')}.csv`)
-      .then((res) => {
-        // window.localStorage.setItem(dateStringKey, res.data);
-        PGELoadProfileStream.write(res.data);
-        if (res.config.url.match(stopFormatted)) {
-          PGELoadProfileStream.end();
-        }
-      })
-      .catch((err) => {
-        dispatch(getPGELoadProfileFailed(err));
-      });
-    // }
+      loadProfileCsvs.set(date.toDate(), localLoadProfile);
+    } else if (date.year() === 2020) { // PGE data in public folder is 2020-only
+      dispatch(getPGELoadProfileStarted());
+      waitingOnAPI = true;
+      ApiClient.get(`/api/pge/${date.format('YYYYMMDD')}.csv`)
+        .then((res) => {
+          window.localStorage.setItem(dateStringKey, res.data);
```

Comment:
Generally I would not recommend writing to localStorage from your actions. I think this logic would be best managed from within a reducer.

Reply:
@krohling I've made this to be a separate case in the reducer called SET_PGE_LOAD_PROFILE